skbio/stats/distance/tests/test_base.py

```diff
@@ -57,6 +64,112 @@ def setUp(self):
                                    np.array(self.dm_2x2_asym_data),
                                    np.array(self.dm_3x3_data)]
 
+    def test_within(self):
+        exp = pd.DataFrame([['a', 'a', 0.0],
+                            ['a', 'c', 4.2],
+                            ['c', 'a', 4.2],
+                            ['c', 'c', 0.0]],
+                           columns=['i', 'j', 'value'])
+        obs = self.dm_3x3.within(['a', 'c'])
+        pdt.assert_frame_equal(obs, exp)
+
+    def test_within_order_stability(self):
+        exp = pd.DataFrame([['a', 'a', 0.0],
+                            ['a', 'c', 4.2],
+                            ['c', 'a', 4.2],
+                            ['c', 'c', 0.0]],
+                           columns=['i', 'j', 'value'])
+
+        # NOTE: order was changed from ['a', 'c'] to ['c', 'a']
+        # but the output order in exp is consistent with
+        # test_within
+        obs = self.dm_3x3.within(['c', 'a'])
+        pdt.assert_frame_equal(obs, exp)
+        obs = self.dm_3x3.within(['a', 'c'])
+        pdt.assert_frame_equal(obs, exp)
+
+    def test_within_missing_id(self):
+        with self.assertRaisesRegex(MissingIDError, "not found."):
+            self.dm_3x3.within(['x', 'a'])
+
+    def test_between(self):
+        exp = pd.DataFrame([['b', 'a', 5.],
+                            ['b', 'c', 6.],
+                            ['b', 'e', 8.],
+                            ['d', 'a', 4.],
+                            ['d', 'c', 6.],
+                            ['d', 'e', 7.]],
+                           columns=['i', 'j', 'value'])
+
+        obs = self.dm_5x5.between(['b', 'd'], ['a', 'c', 'e'])
+        pdt.assert_frame_equal(obs, exp)
+
+    def test_between_order_stability(self):
+        exp = pd.DataFrame([['b', 'a', 5.],
+                            ['b', 'c', 6.],
+                            ['b', 'e', 8.],
+                            ['d', 'a', 4.],
+                            ['d', 'c', 6.],
+                            ['d', 'e', 7.]],
+                           columns=['i', 'j', 'value'])
+
+        # varying the order of the "i" values, result remains consistent
+        # with the test_between result
+        obs = self.dm_5x5.between(['d', 'b'], ['a', 'c', 'e'])
+        pdt.assert_frame_equal(obs, exp)
+
+        # varying the order of the "j" values, result remains consistent
+        # with the test_between result
+        obs = self.dm_5x5.between(['b', 'd'], ['a', 'e', 'c'])
+        pdt.assert_frame_equal(obs, exp)
```

Comment:
like 99% sure this wouldn't be a problem, but it *might* be worth adding another test where both the i and j values are "out-of-order"? e.g.

```python
obs = self.dm_5x5.between(['d', 'b'], ['a', 'e', 'c'])
pdt.assert_frame_equal(obs, exp)
```

(this could be done with minimal effort)